civis/tests/test_io.py

```diff
@@ -206,29 +524,56 @@ def test_read_civis_sql(self, *mocks):
                                        polling_interval=POLL_INTERVAL)
         assert data == expected
 
+    @pytest.mark.dataframe_to_civis
     @pytest.mark.skipif(not has_pandas, reason="pandas not installed")
-    @mock.patch(api_import_str, return_value=civis_api_spec)
-    def test_dataframe_to_civis(self, *mocks):
```

Comment:
I found this older test easier to read, and it checks the result output rather than implementation/call arguments. Can you talk about why these tests were updated?

Reply:
The previous cassette-based tests were difficult to update to account for the use of new endpoints; in particular, re-recording the cassettes proved a challenge, and attempts to learn more about the process of doing so yielded little result. 

This test does seem unwieldy; however, I think part of that is making explicit the stubbing behavior that was previously encapsulated by the cassettes. I am happy to take another pass at simplifying this test to recover some of its prior readability, perhaps by removing some of the mocking around the `DataFrame` and `TemporaryDirectory` in favor of live objects.

Reply:
Ah, got it, this is where we were relying on the cassettes. That makes sense. Where it's possible to test on live objects like dataframes and temp directories, I'd favor that.

Reply:
Refactored - let me know if this helps with readability?